python/ray/_raylet.pyx

```diff
@@ -2649,7 +2649,7 @@ cdef class CoreWorker:
             function_descriptor, specified_cgname)
         coroutine = func(*args, **kwargs)
         if threading.get_ident() == async_thread.ident:
-            future = asyncio.ensure_future(coroutine, eventloop)
+            future = asyncio.ensure_future(coroutine, loop=eventloop)
```

Comment:
I don't know how to trigger this code path given all of our tests fail to uncover the bug.

@simon-mo I saw you added this in #8691, do you know how to trigger this path and why do we need the branching here instead of always calling `run_coroutine_threadsafe`?

Reply:
Hmm i think we should unify the code path here, especially run_coroutine_threadsafe can perform the check for us. 

I believe this is triggered when the serde code gets called while we are in async thread, and the serde functions themselves involves calling async funcs. this is probably due to the fact during `await object_ref` the core worker as of 3 years ago will recursively call into this function so we need to handle this case. 

nevertheless, this is definitely legacy code and safe to unify